Equipo2Hosteleria/src/main/java/Modelo/CarroCompra.java

```diff
@@ -0,0 +1,73 @@
+package Modelo;
+
+import java.util.Arrays;
+
+public class CarroCompra {
+	
+	public double TotalCarro=0;
+	public Object[][] array = new Object[0][4];
+	public ProductoEscogido[] arrayProdEsc = new ProductoEscogido[0];
+	
+	public void anadirProducto(String nombre, double precio, int cantidad) {
+		
+		boolean repetido=false;
+		for(int i=0;i<arrayProdEsc.length;i++) {
+			if(nombre == arrayProdEsc[i].getNombre()) {
+				arrayProdEsc[i].setCantidad(arrayProdEsc[i].getCantidad()+cantidad);
+				arrayProdEsc[i].setPrecioTotal(precio*arrayProdEsc[i].getCantidad());
+				
+				repetido=true;
+			}
+		}
+		if(repetido==false) {
+			aumentarArray();
+			for(int i=0;i<arrayProdEsc.length;i++) {
+				if(arrayProdEsc[i] == null) {
+					arrayProdEsc[i] = new ProductoEscogido();
+					arrayProdEsc[i].setNombre(nombre);
+					arrayProdEsc[i].setCantidad(cantidad);
+					arrayProdEsc[i].setPrecioVenta(precio);
+					arrayProdEsc[i].setPrecioTotal(precio*cantidad);
+					break;
```

Comment:
No hagais break al menos que sea totalmente necesario, porque, puede dar problemas. En un for que tiene un orden de n, es decir, lineal. El agregar un break no va a aportar ninguna mejora en terminos de computo. Porque, el if es una operación que la consideramos atomica. Y, por lo tanto, con el break o sin el tendremos el mismo orden lineal.

Reply:
Ya hemos quitado el break.